08_book_titles

AdrianArcas opened the pull request.

AdrianArcas committed:
Solve 08_book_titles

salcudeann commented on 08_book_titles/book.rb:
```diff
@@ -0,0 +1,24 @@
+$non_capital_words = %w[a the an] + %w[for and nor but or yet] + %w[above across against along among around at before behind 
```
Variables declared using the `$` are global and it is not a good idea to use them. We should better use constants

AdrianArcas committed:
Refactor title= and make  a constant

AdrianArcas committed:
Add attr_reader

salcudeann commented on 08_book_titles/book.rb:
```diff
@@ -0,0 +1,18 @@
+  attr_reader :title
+
+  def title=(title)
+    @title = title.split.map.with_index {|word,i|
```
The general convention is to use curly braces for single-line blocks and `do ... end` for multi-line blocks

AdrianArcas committed:
Change {} with do..end and refactor
Changed {} with do..end because we use multiple lines
Changed "unless" condition to an if because it is not recommended to use
unless with else.
Also changed "or" to || because "or" is not recommended in conditions and
it is used as flowcontrol
